### DISCUSSION TITLE
Outlook Redemption Error

### DISCUSSION THREAD

**Discussion by SilverChinchilla** [cmu-sei/GHOSTS#205] [Q&A]
Howdy Ghost Gang, trying out another Ghosts iteration on a Windows 10 build 19045 and everything is golden except for Outlook. It will Draft and Outbox emails but does not seem to want to put the email addresses into the email so they do not send anywhere.
Logs show a new error I havent seen before so I figured it could be related and any help would be greatly appreciated!

2023/04/04 13:15:46.175|Ghosts.Client.Handlers.Outlook..ctor|RDO load error: System.Runtime.InteropServices.COMException (0x800401FA): Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.IClassFactory.CreateInstance(Object pUnkOuter, Guid& refiid, Object& ppunk)
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject(Guid guid) in Z:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 256
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.new_RDOSession() in Z:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 27

**Comment by sei-dupdyke** [comment]
Hi, what version of Office are you using?

**Reply by SilverChinchilla** [reply]
Thank you for gettin back to me so quickly! I am running 2019 Pro Plus

**Comment by sei-dupdyke** [comment]
Hmm, I test with 2019. Are you having errors with any of the other Office handlers (Word, Excel, PowerPoint)?

**Reply by SilverChinchilla** [reply]
No other errors with Office, just Outlook. It can send an email if I manually type in one of the emails from the Emails-Domain.json, but not when I execute Ghosts

**Reply by sei-dupdyke** [reply]
Do you mind posting up more of the logs prior to this error please? I wanted to see what redemption was doing during the loading of required DLLs.

If you don't want to post here, email them to me directly please.

Otherwise this will be a bear to debug, sorry.

**Reply by SilverChinchilla** [reply]
No issues at all! There were only 2 errors prior to the ones I posted above, I will include the ones above as well:

2023/04/04 13:15:34.888|Ghosts.Client.Infrastructure.Email.EmailContentManager.Check|Email content loaded successfully with 7891 records found
2023/04/04 13:15:37.591|Ghosts.Client.Comms.CheckId.get_Id|config file could not be opened
2023/04/04 13:15:46.175|Ghosts.Client.Handlers.Outlook..ctor|RDO load error: System.Runtime.InteropServices.COMException (0x800401FA): Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.IClassFactory.CreateInstance(Object pUnkOuter, Guid& refiid, Object& ppunk)
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject(Guid guid) in Z:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 256
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.new_RDOSession() in Z:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 27

**Reply by sei-dupdyke** [reply]
Thanks. It was worth a shot. so line 27 is loading a clsid of "29AB7A12-B531-450E-8F7A-EA94C2F3C05F", which led me [to this StackOverflow inquiry](https://stackoverflow.com/questions/21028425/outlook-redemption-issue-with-outlook-32-bit-on-64-bit-machine).

Are you running 64 or 32-bit Office?

**Reply by SilverChinchilla** [reply]
i am running 64-bit office

**Reply by sirbowen78** [reply]
I got a similar problem as yours, I did regsvr32 c:\exercise\ghosts\lib\Redemption.dll, then I restart the win10 machine. the 32 bit should work on all bit of apps.

**Reply by SilverChinchilla** [reply]
thank you @sirbowen78  I will give that a shot and let yall know if it works!

**Reply by SilverChinchilla** [reply]
Quick update, the regsvr32 command did not resolve that error, and the Outlook issue still persists. Still trying a multitude of possible fixes
